### DISCUSSION TITLE
Using isar database files for Multiple apps

### DISCUSSION THREAD

**Discussion by usama-cowlar** [isar/isar#1486] [Help]
Hi, I have come across a use case where I need to share the database files between two flutter applications.

My apps are run on fully controlled custom OS and they both need to share one database file in order to share data.

When I try to do this, I get this exception:
`Unhandled Exception: IsarError: Cannot open Environment: MdbxError (13): Permission denied`

One app creates the database files and the other app uses it. Is there any way to do it with Isar? Please let me know, that'd be of great help.

**Comment by derrickgw** [comment]
There are ways to fix the permissions problem, but that won't fix the bigger picture.  If the apps are running at the same time you will need to make a third app that is a service and has exclusive access to the database.  Synchronizing two different processes to share write access to the same file is _very_ hard.  Modern solutions tend to involve a service that manages the incoming requests.

**Comment by usama-cowlar** [comment]
@derrickgw Thank you so much for the response. The reason for sharing the database file is to share data in a more effective way. Based on the size of data, I'm unable to use Broadcasts or Content providers and I'm using local socket communication with compression mechanism to share data between the apps.

Can you let me know how a third app would benefit in case of my problem? Thank you so much

**Reply by derrickgw** [reply]
I guess you don't have to have a third app. But you do need a single process to own the file. If you are already using sockets to communicate it should be easy.

If you think about a traditional database, you don't interact with the file. You talk over a socket, even on the localhost. Then the database service handles the file IO.
